class.jetpack-gutenberg.php

```diff
@@ -22,29 +37,50 @@
  * @return WP_Block_Type|false The registered block type on success, or false on failure.
  */
 function jetpack_register_block( $slug, $args = array() ) {
+
+	// Attempt to register as Jetpack extension.
+	if ( ! jetpack_register_extension( $slug, $args ) ) {
+		return;
+	}
+
+	// Register Block with Gutenberg.
+	return register_block_type( $slug, $args );
+}
+
+/**
+ * Wrapper function to safely register a gutenberg extension (block or otherwise)
+ *
+ * @param string $slug Slug of the extension.
+ * @param array  $args Arguments that are passed into register_block_type.
+ *
+ * @since X.X.X
+ *
+ * @return Boolean Whether or not the registration was successful.
+ */
+function jetpack_register_extension( $slug, $args = array() ) {
```

Comment:
Does 'extension' only ever refer to Gutenberg extensions in Jetpack? I think that's the case, but otherwise, we might want to make that explicit here. `jetpack_register_editor_extension` or something.